src/hooks/appBadge/useAppBadgeHook.ts

```diff
@@ -0,0 +1,42 @@
+import PushNotification from 'react-native-push-notification';
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+import ChannelList from '../../database/schema/ChannelListSchema';
+import useDatabaseQueueHook from '../core/queue/useDatabaseQueueHook';
+import {DatabaseOperationLabel} from '../../core/queue/DatabaseQueue';
+
+const useAppBadgeHook = () => {
+  const {queue} = useDatabaseQueueHook();
+
+  const updateAppBadgeFromDB = (localDb: SQLiteDatabase) => {
+    if (!localDb) return;
+
+    queue.addJob({
+      operationLabel: DatabaseOperationLabel.AppBadge_UpdateBadgeNumber,
+      id: new Date().valueOf().toString(),
+      task: async () => {
+        try {
+          const allUnreadCount = await ChannelList.getAllUnreadCount(localDb);
+          PushNotification.setApplicationIconBadgeNumber(allUnreadCount || 0);
+        } catch (e) {
+          console.error(e);
+        }
+      }
+    });
+  };
+
+  const udpateAppBadgeWith = (badgeNumber: number) => {
+    try {
+      PushNotification.setApplicationIconBadgeNumber(badgeNumber || 0);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  return {
+    updateAppBadgeFromDB,
+    udpateAppBadgeWith
+  };
+};
+
+export default useAppBadgeHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new hook `useAppBadgeHook` seems to be well implemented. It provides two functions: `updateAppBadgeFromDB` and `udpateAppBadgeWith`. The first one updates the application badge number based on the unread count from the database, while the second one directly sets the application badge number with a provided value.

However, there's a minor typo in the function name `udpateAppBadgeWith`, which should be corrected to `updateAppBadgeWith`.

```diff
-  const udpateAppBadgeWith = (badgeNumber: number) => {
+  const updateAppBadgeWith = (badgeNumber: number) => {
```

Also, error handling could be improved by providing more context about where the error occurred. Instead of just logging the error, it would be helpful to include some additional information about the operation that failed.

```diff
- console.error(e);
+ console.error("Error updating app badge from DB: ", e);

- console.error(e);
+ console.error("Error updating app badge with provided number: ", e);
```
Lastly, it's important to note that this hook relies on the `react-native-push-notification` library for setting the application badge number. This library might not support all platforms or versions equally, so it's crucial to ensure that the target platforms of your application are supported.


<!-- This is an auto-generated comment by OSS CodeRabbit -->